### DISCUSSION TITLE
Changing max temp values on the BTT TFT35 Touch screen mode

### DISCUSSION THREAD

**Discussion by amv101** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1567] [General]
I spent hours trying to figure this out, so I wanted to post the solution if anyone else is having an issue setting the hotend max temp in the touchscreen mode of the BTT TFT35.

You have to edit the config.ini file (max_temp:) and the configuration.h file (#define HEAT_MAX_TEMP). Also make sure your mainboard firmware has been modified as well. I used Visual Studio and PIO to edit and compile the new firmware. You can find the how-to here. https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware

What took me a while to find, as I didnt see it posted anywhere, was needing to also edit the config.h file in TFT/src > API and modify #define MAX_TOOL_TEMP. It is set at 300, so you will get an error when trying to load the firmware on the TFT35 if your config,ini is set higher than 300, without this change. Something like "Invalid Values" and it shows the bad code.

Crazy that you have to change the max temp in 3 places on the touchscreen mode.

Hope this helps someone else!

**Comment by oldman4U** [comment]
Hi. You have to change only the config.ini which you copy on the sd card used for updating the firmware. Like described in the ReadMe on the main page of this repository.

Happy printing

**Reply by amv101** [reply]
Wont work unless you change the line below in config.h. Even says right there on the line, that temps above MAX_TOOL_TEMP will not be parsed. Nothing I did worked, until I changed this to 500. Now I can set my Extruder to 500C. Before that, it wouldn't go above 300C. Dont know what to tell you.

Line 155  #define MAX_TOOL_TEMP   300   // extruder temp over this will not pe parsed.

**Reply by kisslorand** [reply]
Exactly!
In config.h:
`#define MAX_TOOL_TEMP             300          // extruder temp over this will not be parsed.`
Hardcoded limits were already mentioned in BR #1524
Minimum is 25°C, maximum is set to 300°C.

**Comment by oldman4U** [comment]
I see. So this should be mentioned in config.ini.

Still not sure what the 3 files are which have to be changed. Maybe it should be mentioned in all of them.

**Comment by bigtreetech** [comment]
Is it enough to limit maxtemp to 500℃ in firmware?

**Reply by oldman4U** [reply]
What is the limit good for?

**Reply by amv101** [reply]
I do not need 500C specifically. I just upgraded my hotend to allow higher temps, and when I went to test the max temp, it wouldnt go above 300C. It seems most of the high temp hotends are rated up to 500C.

Maybe hardcode it up to 500C and let the user edit the config.ini, which defaults to 275C.

**Reply by oldman4U** [reply]
Still not sure why there has to be a hardcoded limit but in case it has to be there I would recommend to use a value which is higher than what is possible today. 500 sounds good

**Comment by oldman4U** [comment]
Please upload official v27 release to see if it works. 

Please help the community and close the discussion once you do not need it anymore. Thank you
